spec/models/concerns/date_helpers_spec.rb

```diff
@@ -0,0 +1,29 @@
+require 'rails_helper'
+
+RSpec.describe DateHelpers do
+  let(:test_class) do
+    Class.new { include DateHelpers }
+  end
+
+  subject { test_class.new }
+
+  context 'valid_date?' do
+    it 'method should be defined' do
+      expect(subject.private_methods.include? :valid_date?).to eq true
+    end
+
+    it 'should return true for a valid date' do
+      [Date.current, "10-10-2012"].each do |good_date_value|
+        expect(subject.send(:valid_date?, good_date_value)).to eq true
```

Comment:
good and bad dates are tested directly on the method so we don't have to test good and bad dates everywhere this method is used. We just check if this method is called or not.